mathics/builtin/arithmetic.py

```diff
@@ -1477,10 +1536,14 @@ class Gamma(_MPMathMultiFunction):
     #> N[Gamma[24/10], 100]
      = 1.242169344504305404913070252268300492431517240992022966055507541481863694148882652446155342679460339
     #> N[N[Gamma[24/10],100]/N[Gamma[14/10],100],100]
-     = 1.4
+     = 1.400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
     #> % // Precision
      = 100.
 
+    #> Gamma[1.*^20]
+     : Overflow occurred in computation.
+     = Overflow[]
```

Comment:
It took me a few tries to figure out how to do this:
```
In[36]:= Gamma[1.0000000000000*^20]
 : Overflow occurred in computation.
Out[36]= Overflow[]

In[37]:= Precision[1.0000000000000*^20]
Out[37]= MachinePrecision

In[38]:= N[MachinePrecision]
Out[38]= 15.9546

In[39]:= Gamma[1.`16*^20]
Out[39]= 1.932849514310098*^1956570551809674817225

In[40]:= Precision[%]
Out[40]= 16.
```
Since this brings us much closer to the MMA behaviour, is it worth adding some documentation to explain how overflows work and how to input numbers with different precisions? I don't think it should hold this PR up, but it'd be nice to have.